crates/ra_hir/src/ty/tests.rs

```diff
@@ -20,6 +20,42 @@ use crate::{
 // against snapshots of the expected results using insta. Use cargo-insta to
 // update the snapshots.
 
+#[test]
+fn infer_try() {
+    let (mut db, pos) = MockDatabase::with_position(
+        r#"
+//- /main.rs
+enum Result<O, E> {
+    Ok(O),
+    Err(E)
+}
+
+impl<O, E> ::std::ops::Try for Result<O, E> {
```

Comment:
Cosmetics, but seems slightly better to keep `enum Result` and this `impl` in the `lib.rs` mock stdlib file.

Hm, while we are at it, perhaps it's better to name `lib.rs` as `std.rs` in this and other tests, where it plays the role of "mock" stdlib?